pkg/app/app_fetch.go

```diff
@@ -43,6 +44,31 @@ func (a *App) fetch(dstPath string) (string, exec.CmdRunResult) {
 	}
 
 	result = a.runVendir(confReader, dstPath)
+	// retry if error occurs before reporting failure.
+	// This is mainly done to support private registry
+	// authentication for images/bundles since placeholder
+	// secrets may not be populated in time.
+	if result.Error != nil && a.HasImageOrImgpkgBundle() {
+		// Only retrying once resulted in flaky behavior
+		// for private auth so use 5 iterations.
+		for i := 0; i < 5; i++ {
```

Comment:
TODO: figure out why sg populating is that slow that we need 5s

Reply:
This probably does not actually take 5s. Five iterations was chosen arbitrarily.

Reply:
Added to #336. In a simple case, we observed there can be a 3s difference between PackageInstall being created and placeholder secret being populated.